feat(build): setup multiarch build for arm64

Darkbladecr opened the pull request:
According to issue #1160 #1138 

I do not know what the CI environment is for building the docker images. But I have updated the Dockerfile to be able to accommodate multiarch builds. Note my environment is `arm64` so I have to set up the docker for amd64, the converse may be true for your build CI environment.

As per the `multi_arch` script the docker buildx environment needs to be setup:

```bash
docker context create container
docker buildx create --use container
```

I assume the CI is a linux environment in which case buildx needs to be installed: https://github.com/docker/buildx?tab=readme-ov-file#linux-packages

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/heroiclabs/nakama?pullRequest=1161) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/heroiclabs/nakama?pullRequest=1161) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/heroiclabs/nakama?pullRequest=1161) it.</sub>

mofirouz commented:
Thanks for this. Do you have a way to update the build README for binary releases too?

Darkbladecr commented:
> Thanks for this. Do you have a way to update the build README for binary releases too?

I haven't touched the binary workflow, so they should work as setup previously. I wanted to try and avoid any breaks with existing code or CI/CD. If you want me to clean everything up so that it works with this single Dockerfile, and update the readme for how to run it for any platform with `docker buildx` I can do that?

Darkbladecr commented:
Removed duplicate docker builds for both the main server and the pluginbuilder
Added info on how to use/setup in the [README.md](https://github.com/Darkbladecr/nakama/blob/master/build/README.md)

Updated the scripts to automatically pull the latest tag and info:

```bash
VER="$(git tag -l --sort=-creatordate | head -n 1)"
DOCKER_VER=heroiclabs/nakama:${VER:1}
echo Building $DOCKER_VER
docker buildx build --platform linux/amd64,linux/arm64 \
  --builder=container \
  --build-arg repo=https://github.com/heroiclabs/nakama.git \
  --build-arg commit="$(git rev-parse --short HEAD)" \
  --build-arg version=$VER \
  --tag $DOCKER_VER \
  --push .
```

Darkbladecr commented:
Anything else that needs to be done here @mofirouz ?

tobyw121 commented:
hello , please have a look at my Github : https://github.com/tobyw121/nakama_docker_aarch64/tree/main

I have created a docker for a Nakama on aarch64 here .

sesposito commented:
Hello @Darkbladecr,

We've now resolved this issue on https://github.com/heroiclabs/nakama/pull/1316. I'd like to thank you for your contribution regardless, it was helpful to experiment on how to best achieve it, from my experimentation emulation turned out to be more reliable than cross-compilation so we went with that approach instead.

Best.
